xbmc/guilib/handlers/sources/GUISourcesAnnouncementHandler.cpp

```diff
@@ -0,0 +1,42 @@
+/*
+ *  Copyright (C) 2024 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "GUISourcesAnnouncementHandler.h"
+
+#include "GUIUserMessages.h"
+#include "ServiceBroker.h"
+#include "guilib/GUIComponent.h"
+#include "guilib/GUIWindowManager.h"
+#include "interfaces/AnnouncementManager.h"
+
+CGUISourcesAnnouncementHandler::CGUISourcesAnnouncementHandler()
+{
+  CServiceBroker::GetAnnouncementManager()->AddAnnouncer(this);
+}
+
+CGUISourcesAnnouncementHandler::~CGUISourcesAnnouncementHandler()
+{
+  CServiceBroker::GetAnnouncementManager()->RemoveAnnouncer(this);
+}
+
+void CGUISourcesAnnouncementHandler::Announce(ANNOUNCEMENT::AnnouncementFlag flag,
+                                              const std::string& sender,
+                                              const std::string& message,
+                                              const CVariant& data)
```

Comment:
Do you really need the flexibility of a `CVariant`? Or would it always be the path passed in here?

Reply:
This is just an example of the implementation, I am pretty sure there'll be others that require us to pass other types rather than only a string. I consider this a bit out of the scope of the purposed PR - unfortunately we can't tackle everything at once :/

Reply:
Yeah this is something I can improve, since I can implicitly create a string `CVariant`. Will adjust that.

Reply:
>  unfortunately we can't tackle everything at once :/

It was worth a try. :-)